app/scripts/services/discovery.js

```diff
@@ -1,11 +1,15 @@
 'use strict';
 
 angular.module('openshiftConsole')
-  .factory('APIDiscovery', ['LOGGING_URL', 'METRICS_URL', '$q', function(LOGGING_URL, METRICS_URL, $q) {
+  .factory('APIDiscovery', ['LOGGING_URL', 'METRICS_URL', '$q', '$filter', function(LOGGING_URL, METRICS_URL, $q, $filter) {
     return {
       // Simulate asynchronous requests for now. If these are ever updated to call to a discovery
       // endpoint, we need to make sure to trigger a digest loop using (or update all callers).
-      getLoggingURL: function() {
+      getLoggingURL: function(project) {
+        if(project.metadata.annotations[$filter('annotationName')('loggingUIHostname')]) {
+           return 'https://' + project.metadata.annotations[$filter('annotationName')('loggingUIHostname')];
+        }
```

Comment:
I think this is easier to read (untested)

```js
var loggingUIHostname = $filter('annotation')(project, 'loggingUIHostname');
if (loggingUIHostname) {
  return 'https://' + loggingUIHostname;
}
```

Reply:
agree.